src/tag.c

```diff
@@ -2247,179 +2267,159 @@ find_tags(
 		}
 
 		/*
-		 * Add the found match in ga_match[mtt], avoiding duplicates.
+		 * Add the found match in ht_match[mtt].
 		 * Store the info we need later, which depends on the kind of
 		 * tags we are dealing with.
 		 */
-		if (ga_grow(&ga_match[mtt], 1) == OK)
-		{
-		    int len;
-		    int heuristic;
+		int len;
+		int heuristic;
```

Comment:
Thanks for the feedback!  I initially had those in a nested block (just removed the if condition).  Now that I've removed that, I do need to move the declarations.

As far as standard compatibility, it might be worth adding a CI build that uses `-std=gnu89 -pedantic` to more easily find these issues.  I notice that's not the only issue in the current source.